src/main/java/com/pitchain/common/exception/ExceptionHandlerFilter.java

```diff
@@ -0,0 +1,39 @@
+package com.pitchain.common.exception;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.pitchain.common.apiPayload.dto.CustomApiResponse;
+import com.pitchain.common.apiPayload.dto.ResponseDTO;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@Component
+public class ExceptionHandlerFilter extends OncePerRequestFilter {
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        try {
+            filterChain.doFilter(request, response);
+        } catch (RuntimeException e) {
+            GeneralHandler generalHandler = (GeneralHandler) e;
+            ResponseDTO error = generalHandler.getError();
+            setResponse(response, error);
+        }
```

Comment:
타입 캐스팅 부분에서 에러가 발생할 가능성이 존재해 보입니다
다음과 같이 수정하면 좋을 것 같아요
```suggestion
        } catch (RuntimeException e) {
            if (e instanceof GeneralHandler generalHandler) {
              ResponseDTO error = generalHandler.getError();
              setResponse(response, error);
            }
        }
```